add fallback screen reader styles

haszari opened the pull request:
Fixes #3265 

This PR adds styling for screen-reader text elements. Usually this is handled by the theme. However if the theme doesn't have support the screen reader text may be rendered inline (which is not intended); these fallback styles ensure screen reader text is hidden by default, and provides `:focus` styles for when the text has focus and is being read.

The specifics of the styles are pasted direct from underscores, with some colours changed. Feedback welcome on how to generalise these so they are a minimal, safe fallback (i.e. should not be opinionated).

#### Accessibility

- [ ] I've tested using only a keyboard (no mouse)
- [ ] I've tested using a screen reader
- [ ] All animations respect [`prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion)
- [ ] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)
- [ ] I've updated [this doc](../docs/blocks/feature-flags-and-experimental-interfaces.md) for any feature flags or experimental interfaces implemented in this pull request.

I've tested with Safari VoiceOver but I couldn't focus the element in normal use. Keen for some testing of this with various browsers/platforms and screen reader tech :)

### Screenshots

Focused in chrome and safari:

<img width="519" alt="Chrome" src="https://user-images.githubusercontent.com/4167300/102292375-e477a000-3fa9-11eb-9f68-08928b1a39c0.png">

<img width="445" alt="Safari" src="https://user-images.githubusercontent.com/4167300/102292382-e6d9fa00-3fa9-11eb-81fd-e207535b9499.png">

Notes: 

- I faked `:focus` using browser dev tools.
- When unfocused, screen reader text should not be visible.

For comparison, here's a screenshot of the issue:

<img width="275" alt="Screen Shot 2020-12-16 at 2 27 46 PM" src="https://user-images.githubusercontent.com/4167300/102292889-e9891f00-3faa-11eb-93b1-52b506248965.png">

### How to test the changes in this Pull Request:
1. Install a theme that doesn't provide screen reader styles. I built a new `Bad Theme` using [Underscores](https://underscores.me) and deleted the screen reader styles.
2. Add checkout block to a page and stuff to cart.
3. View checkout on front end - confirm `.screen-reader-text` is hidden by default and is helpful when using a screen reader.
4. Test with other themes to ensure these styles don't conflict or cause issues.

### Changelog

> Fix: Added fallback styling for screen reader text.

haszari committed:
add fallback screen reader styles

Aljullu reviewed (approved):
> I've tested with Safari VoiceOver but I couldn't focus the element in normal use. Keen for some testing of this with various browsers/platforms and screen reader tech :)

If you were using the <kbd>Tab</kbd> key, I think that's to be expected. Screen readers usually provide a different navigation method which is more "line-by-line" so you can navigate to non-focusable items. With [Orca](https://help.gnome.org/users/orca/stable/) it's done with the arrow keys, I guess it's probably the same in other screen readers?

I tested removing the `screen-reader-text` styles from Storefront and I can confirm this PR adds a fallback so screen reader elements are hidden.

In order to test `:focus` styles, I converted the `<FormStep>` `legend` element into an `a` and it looked great:

![imatge](https://user-images.githubusercontent.com/3616980/102329082-38718d00-3f88-11eb-999c-11b6813e2dfc.png)

My only concern is that we are introducing a new set of 'screen-reader' styles, which differs from the ones in other projects:
  * [WooCommerce](https://github.com/woocommerce/woocommerce/blob/16faf915e0055e48d4f0c2f647696716df957bbb/assets/css/woocommerce.scss#L41-L48)
  * [WooCommerce Blocks mixin](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/assets/css/abstracts/_mixins.scss#L53-L64)
  * [Storefront](https://github.com/woocommerce/storefront/blob/trunk/assets/css/base/_base.scss#L507-L531)

Not blocking merge, but do you think we could use one of those :point_up_2: as the base instead of introducing a new one from underscores? :slightly_smiling_face:

haszari committed:
use shared blocks screen-reader styles from mixin:
- tweak mixin to align with focus-reveal style
- add new mixin for focus-reveal style, since it is coupled to
visually-hidden mixin

haszari commented:
> My only concern is that we are introducing a new set of 'screen-reader' styles … we could use one of those 👆

Good thinking @Aljullu , I should have looked more carefully for prior art!

I've switched to using our (most local) `visually-hidden()` mixin. I also made a new mixin for the companion focus style, since they are coupled.

Aljullu reviewed (approved):
Nice to see a new mixing for the focus state. I didn't re-test it but code changes look good. :shipit:
